### PR TITLE
Ubuntu/focal

### PR DESCRIPTION
**Pull request by TheRealFalcon**
new upstream snapshot + last two commits

### CODE REVIEW

File: debian/changelog [hunk 1]
```diff
@@ -26,8 +29,14 @@ cloud-init (24.1.3-0ubuntu1~20.04.4) UNRELEASED; urgency=medium
     - d/p/status-retain-recoverable-error-exit-code.patch
     - d/p/revert-551f560d-cloud-config-after-snap-seeding.patch
   * Upstream snapshot based on upstream/main at db859451.
+  * refresh patches:
```
Comment (holmanb) [inline debian/changelog:32]: second section unwanted

File: debian/changelog [hunk 2]
```diff
@@ -26,8 +29,14 @@ cloud-init (24.1.3-0ubuntu1~20.04.4) UNRELEASED; urgency=medium
...
+    - d/p/cli-retain-file-argument-as-main-cmd-arg.patch
+    - d/p/drop-unsupported-systemd-condition-environment.patch
```
Comment (holmanb) [inline debian/changelog:34]: these three are duplicates

Comment (TheRealFalcon) [reply debian/changelog]: Updated. Also used your name/description for the patch.

**Review by holmanb — COMMENTED** [review]
a couple of nits about the changelog for release but otherwise looks good

File: debian/changelog [hunk 3]
```diff
@@ -16,7 +16,12 @@ cloud-init (24.1.3-0ubuntu1~20.04.4) UNRELEASED; urgency=medium
     those keys and emits warnings at systemctl status
   * d/p/revert-551f560d-cloud-config-after-snap-seeding.patch: retain systemd
     ordering cloud-config.service After=snapd.seeded.service
+  * d/p/add-deprecation-info-boundary.patch: Update
+    DEPRECATION_INFO_BOUNDARY to ensure new deprecations don't trigger
+    warnings.
   * refresh patches:
+    - d/p/cli-retain-file-argument-as-main-cmd-arg.patch
+    - d/p/drop-unsupported-systemd-condition-environment.patch
```
Comment (holmanb) [inline debian/changelog:24]: we don't need this line, since we added this patch in this release

**Pull request merged**